File: boj/boj-10828.py
import sys
input = sys.stdin.readline

N = int(input().rstrip())
stack = []

for _ in range(N):
    string = list(input().split())
    command = ''
    num = 0
    
    if len(string) == 1:
        command = string[0]
    else:
        command, num = string
    
    if command == 'push':
        stack.append(num)
    elif command == 'pop':
        if len(stack) == 0:
            print(-1)
        else:
            print(stack.pop(-1))
    elif command == 'size':
        print(len(stack))
    elif command == 'empty':
        if len(stack) == 0:
            print(1)
        else:
            print(0)
    elif command == 'top':
        if len(stack) == 0:
            print(-1)
        else:
            print(stack[-1])

File: boj/swift/boj-10828.swift
import Foundation

let n = Int(readLine()!)!
var stack: [Int] = []

for _ in 0..<n {
    let cn = readLine()!.split(separator: " ")
    let command = cn[0]
    let num = (cn.count == 2) ? Int(cn[1]) : nil
    
    switch command {
    case "push":
        stack.append(num!)
    case "pop":
        print(stack.popLast() ?? -1)
    case "size":
        print(stack.count)
    case "empty":
        _ = (stack.count == 0) ? print(1) : print(0)
    case "top":
        print(stack.last ?? -1)
    default:
        continue
    }
}

